fairlearn/metrics/_plotter.py

```diff
@@ -0,0 +1,297 @@
+
+    kind : str
+        The type of plot to display. e.g. "bar", "line", etc.
+        List of options is detailed in :meth:`pandas.DataFrame.plot`
+
+    subplots : bool
+        Whether or not to plot metrics on separate subplots
+
+    legend_label : str
+        The label corresponding to the confidence interval bars
+
+    df_all_errors : pd.DataFrame, default=None
+        Optional argument for DataFrame that indicates the errors to plot
+
+    \*\*kwargs
+        Keyword arguments that are passed in to :meth:`pandas.DataFrame.plot`.
+
+    Returns
+    -------
+    :class:`matplotlib.axes.Axes` or :class:`numpy.ndarray` of them
+    """
+    if df_all_errors is not None:
+        yerr = np.array(
+            [
+                np.array([[*row] for row in df_all_errors[metric]]).T
+                for metric in metrics
+            ]
+        )
+        if kind == "scatter":
+            axs = df[metrics].plot(
+                linestyle="",
+                marker="o",
+                yerr=yerr,
+                subplots=subplots,
+                **kwargs
+            )
+        else:
+            axs = df[metrics].plot(
+                kind=kind, yerr=yerr, subplots=subplots, **kwargs
+            )
+
+        if isinstance(axs, np.ndarray):
+            for ax in axs.flatten():
+                _build_legend(ax, kind, legend_label)
+        else:
+            _build_legend(axs, kind, legend_label)
+
+    else:
+        if kind == "scatter":
+            axs = df[metrics].plot(
+                linestyle="", marker="o", subplots=subplots, **kwargs
+            )
+        else:
+            axs = df[metrics].plot(kind=kind, subplots=subplots, **kwargs)
+
+    return axs
+
+
+def plot_metric_frame(
+    metric_frame: MetricFrame,
+    *,
+    kind: str = "scatter",
+    metrics: Union[List[str], str] = None,
+    conf_intervals: Union[List[str], str] = None,
+    subplots: bool = True,
+    plot_error_labels: bool = False,
+    error_labels_precision: int = 4,
+    error_labels_fontsize: int = 8,
+    error_labels_color: str = "black",
+    error_labels_ha: str = "center",
+    legend_label: str = "95% CI",
+    **kwargs
+):
+    r"""Visualization for metrics with and without confidence intervals.
+
+    Plots a given metric and its given error (as described by `conf_intervals`)
+
+    This function takes in a :class:`fairlearn.metrics.MetricFrame` with precomputed metrics
+    and metric errors and a `conf_intervals` array to interpret the columns
+    of the :class:`fairlearn.metrics.MetricFrame`.
+
+    The items at each index of the given `metrics` array and given `errors` or `conf_intervals`
+    array should correspond to a pair of the same metric and metric error, respectively.
+
+    Parameters
+    ----------
+    metric_frame : fairlearn.metrics.MetricFrame
+        The collection of disaggregated metric values, along with the metric errors.
+
+    kind : str, default="scatter"
```

Comment:
Changed to 'point'

Reply:
Actually, @MiroDudik it seems that "point" isn't valid for `DataFrame.plot()`:
https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.plot.html
Reverting back to 'scatter'

Reply:
The fact that 'point' is not valid is actually a good thing, because the goal here is to add `kind='point'` as a new kind of a plot (that happens to be implemented using the scatter plot functionality with some tweaks). So I think we should still switch to `'point'`.

Reply:
So you're saying that this should be a string whose only valid value is `point` which we rewrite to `scatter` when invoking the `DataFrame` plot?